### PR TITLE
Issue1149 - Document EntityRepository to help contributors understand how to use it to add new entities

### PR DESCRIPTION
**Pull request by sureshms**
### Describe your changes :

Document EntityRepository to help contributors understand how to use it to add new entities. See #1149 for more details

#
### Type of change :

- [] Bug fix
- [] Improvement
- [] New feature
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@pmbrull

### CODE REVIEW

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/EntityRepository.java [hunk 1]
```diff
@@ -328,13 +422,19 @@ public EntityUpdater(T original, T updated, boolean patchOperation) {
...
+    public final void update() throws IOException, ParseException {
       updated.setId(original.getId());
       updateDescription();
       updateDisplayName();
       updateOwner();
       updateTags(updated.getFullyQualifiedName(), "tags", original.getTags(), updated.getTags());
       entitySpecificUpdate();
+
+      // Store the updated entity
+      store();
```
Comment (pmbrull) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/EntityRepository.java:437]: May this call be `storeEntity`?

**Review by pmbrull — APPROVED** [review]
Amazing work @sureshms, really appreciate it 🙏 

Just layed out a comment to see if I am missing something.

Thanks!

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/EntityRepository.java (hunk 1 above)
Comment (sureshms) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/EntityRepository.java:437]: This an internal method of `EntityUpdater` diff from `EntityRepository`. We can change the name, if necessary.

Comment (pmbrull) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/EntityRepository.java:437]: What would you think of `storeUpdates`?

Comment (sureshms) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/EntityRepository.java:445]: Good idea. I will change it to `storeUpdate`.

**Pull request merged**